File: eukay/lib/pages/cart/mappers/cart_model.dart
class CartModel {
  final String id, productImage, productName, productId;
  final double price, subTotal;
  final int quantity;

  CartModel({
    required this.id,
    required this.productId,
    required this.productImage,
    required this.productName,
    required this.price,
    required this.quantity,
    required this.subTotal,
  });

  factory CartModel.fromJson(Map<String, dynamic> json) {
    return CartModel(
      id: json['_id'],
      productId: json['productId'],
      productImage: json['productImage'],
      productName: json['productName'],
      price: (json["price"] is int)
          ? (json["price"] as int).toDouble()
          : json["price"].toDouble(),
      subTotal: (json["subTotal"] is int)
          ? (json["subTotal"] as int).toDouble()
          : json["price"].toDouble(),
      quantity: json['quantity'],
    );
  }
}

File: server/models/cart.model.js
const mongoose = require("mongoose");

const CartSchema = mongoose.Schema(
  {
    productImage: {
      type: String,
      required: true,
    },

    productName: {
      type: String,
      required: true,
    },

    quantity: {
      type: Number,
      required: true,
    },

    price: {
      type: Number,
      required: true,
    },

    subTotal: {
      type: Number,
      required: true,
    },

    toCheckOut: {
      type: Boolean,
      required: false,
      default: false,
    },

    size: {
      type: String,
      required: true,
    },

    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },

    productId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Product",
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

const Cart = mongoose.model("Cart", CartSchema);

module.exports = Cart;
